pkg/operator/controller/machineconfig/controller.go

```diff
@@ -19,96 +19,29 @@ package machineconfigcontroller
-	CtrlConfig     *v1alpha1.NodeObservabilityMachineConfig
-	EventRecorder  record.EventRecorder
-	PrevSyncChange map[string]PrevSyncData
-}
-
-// PrevSyncData is for storing the config changes made in
-// previous reconciliation and the config used.
-type PrevSyncData struct {
-	action string
-	config interface{}
-}
-
-const (
-	finalizer = "NodeObservabilityMachineConfig"
-
-	defaultRequeueTime = 30 * time.Minute
-
-	// MCAPIVersion is the machine config API version
-	MCAPIVersion = "machineconfiguration.openshift.io/v1"
-
-	// MCKind is the machine config resource kind
-	MCKind = "MachineConfig"
-
-	// MCPoolKind is the machine config pool resource king
-	MCPoolKind = "MachineConfigPool"
-
-	// ProfilingMCPName is the name of MCP created for
-	// CRI-O, Kubelet... machine configs by this controller
-	ProfilingMCPName = "nodeobservability"
-)
-
-var (
-	// ProfilingMCSelectorLabels is for storing the labels to
-	// match with profiling MCP
-	ProfilingMCSelectorLabels = map[string]string{
-		"machineconfiguration.openshift.io/role": ProfilingMCPName,
-	}
-
-	// NodeSelectorLabels is for storing the labels to
-	// match the nodes to include in MCP
-	NodeSelectorLabels = map[string]string{
-		"node-role.kubernetes.io/worker": "",
-	}
-
-	// MachineConfigLabels is for storing the labels to
-	// add in machine config resources
-	MachineConfigLabels = map[string]string{
-		"machineconfiguration.openshift.io/role":                      ProfilingMCPName,
-		"machineconfigs.nodeobservability.olm.openshift.io/profiling": "",
-	}
-)
-
 //+kubebuilder:rbac:groups=nodeobservability.olm.openshift.io,resources=nodeobservabilitymachineconfigs,verbs=get;list;watch;create;update;patch;delete
 //+kubebuilder:rbac:groups=nodeobservability.olm.openshift.io,resources=nodeobservabilitymachineconfigs/status,verbs=get;update;patch
 //+kubebuilder:rbac:groups=nodeobservability.olm.openshift.io,resources=nodeobservabilitymachineconfigs/finalizers,verbs=update
-//+kubebuilder:rbac:groups=machineconfiguration.openshift.io,resources=machineconfigs,verbs=get;list;watch;create;update;patch;delete
-//+kubebuilder:rbac:groups=machineconfiguration.openshift.io,resources=kubeletconfigs,verbs=get;list;watch;create;update;patch;delete
-//+kubebuilder:rbac:groups=machineconfiguration.openshift.io,resources=machineconfigpools,verbs=get;list;watch;create;update;patch;delete
+//+kubebuilder:rbac:groups=machineconfiguration.openshift.io,resources=machineconfigs,verbs=get;list;create;delete
+//+kubebuilder:rbac:groups=machineconfiguration.openshift.io,resources=machineconfigpools,verbs=get;list;create;delete
+//+kubebuilder:rbac:groups=core,resources=nodes,verbs=get;list;patch
 
 // Reconcile is part of the main kubernetes reconciliation loop which aims to
 // move the current state of the cluster closer to the desired state.
 //
 // For more details, check Reconcile and its Result here:
 // - https://pkg.go.dev/sigs.k8s.io/controller-runtime@v0.11.0/pkg/reconcile
```

Comment:
this comment is auto-generated, pls add a more specific doc text to this controller

Reply:
comment incorporated